src/api/app/controllers/webui/package_controller.rb

```diff
@@ -182,24 +182,22 @@ def statistics
                                                       architecture: params[:arch]).results
   end
 
+  # FIXME: This is Webui::Packages::BinariesController#show
   def binary
-    @package_name = params[:package]
+    @package_name = params[:package] # might be a multibuild flavor
     # Ensure it really is just a file name, no '/..', etc.
     @filename = File.basename(params[:filename])
 
-    @fileinfo = Backend::Api::BuildResults::Binaries.fileinfo_ext(@project, @package_name, @repository.name, @arch.name, @filename)
+    @fileinfo = Backend::Api::BuildResults::Binaries.fileinfo_ext(@project, @package_name, @repository.name, @architecture.name, @filename)
     raise ActiveRecord::RecordNotFound, 'Not Found' unless @fileinfo
 
-    url_generator = ::PackageControllerService::URLGenerator.new(project: @project, package: @package_name,
-                                                                 user: User.possibly_nobody, arch: @arch,
-                                                                 repository: @repository, filename: @filename)
-    @download_url = url_generator.download_url_for_file_in_repo
-
-    logger.debug "accepting #{request.accepts.join(',')} format:#{request.format}"
-    # little trick to give users eager to download binaries a single click
-    redirect_to(@download_url) && return if request.format != Mime[:html] && @download_url
+    @download_url = @package.download_url_for_binary(repository_name: @repository.name,
+                                                     architecture_name: @architecture.name,
+                                                     package_name: @package_name,
+                                                     filename: @filename)
   end
 
+  # FIXME: This is Webui::Packages::BinariesController#index
```

Comment:
I don't understand this comment, can you explain a bit?

Reply:
This should be extracted into it's own controller...